src/v/hashing/tests/hash_bench.cc

```diff
@@ -11,64 +11,259 @@
 #include "hashing/fnv.h"
 #include "hashing/twang.h"
 #include "hashing/xx.h"
+#include "model/fundamental.h"
+#include "model/ktp.h"
+#include "model/namespace.h"
 #include "random/generators.h"
 
 #include <seastar/core/reactor.hh>
+#include <seastar/core/sstring.hh>
 #include <seastar/testing/perf_tests.hh>
 
+#include <absl/hash/hash.h>
+#include <absl/strings/string_view.h>
 #include <boost/crc.hpp>
 
 static constexpr size_t step_bytes = 57;
 
-PERF_TEST(boost_crc16_fn, header_hash) {
+static constexpr size_t inner_iters = 1000;
+
+template<typename F>
+static size_t header_body(F n) {
     auto buffer = random_generators::gen_alphanum_string(step_bytes);
     perf_tests::start_measuring_time();
-    boost::crc_16_type crc;
-    crc.process_bytes(buffer.data(), buffer.size());
-    auto o = crc.checksum();
-    perf_tests::do_not_optimize(o);
+    for (auto i = inner_iters; i--;) {
+        auto s = n(buffer);
+        perf_tests::do_not_optimize(s);
+    }
```

Comment:
The existing hash tests were returning meaningless results because the overhead of timing is like 800 nanoseconds (due to getting instruction count) while many of these  methods are like 10s of nanos. So everything looked the same.

So introduce an inner loop with 1000s of iterations to actually measure the hash functions.